File: Core/DataFilter/CS/DataFilterExplorer/Samples/Controls/FlexChartDateTimeFilterView.cs
using C1.Chart;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace DataFilterExplorer.Samples
{
    public partial class FlexChartDateTimeFilterView : FlexChartFilterView
    {
        #region fields
        private ComboBox _cbGroupBy, _cbAggregate;
        private DateTimeGroupOption _groupBy;
        private AggregateFunction _aggregate;
        #endregion

        #region Properties
        public DateTimeGroupOption GroupBy
        {
            get { return _groupBy; }
            set
            {
                if (_groupBy != value)
                {
                    _groupBy = value;
                    _cbGroupBy.SelectedIndex = _cbGroupBy.SelectedIndex == -1 ? -1 : (int)value;
                    GroupByChanged?.Invoke(this, null);
                }
            }
        }
        public AggregateFunction Aggregate
        {
            get { return _aggregate; }
            set
            {
                if (_aggregate != value)
                {
                    _aggregate = value;
                    _cbAggregate.SelectedIndex = _cbAggregate.SelectedIndex == -1 ? -1 : (int)value;
                    AggregateChanged?.Invoke(this, null);
                }
            }
        }
        public bool ShowGroupBy
        {
            get { return _cbGroupBy.Visible; }
            set { _cbGroupBy.Visible = value; }
        }
        public bool ShowAggregate
        {
            get { return _cbAggregate.Visible; }
            set { _cbAggregate.Visible = value; }
        }
        #endregion

        #region c'tor
        public FlexChartDateTimeFilterView(IEnumerable<object> items, string bindingX, string binding, ChartType chartType = ChartType.Column) : base(items, bindingX, binding, chartType)
        {
            InitializeComponent();

            _cbGroupBy = new ComboBox()
            {
                Width = 80,
                FlatStyle = FlatStyle.Flat,
                DataSource = Enum.GetValues(typeof(DateTimeGroupOption)),
            };
            _cbGroupBy.SelectedIndexChanged += _cbGroupBy_SelectedIndexChanged;

            _cbAggregate = new ComboBox()
            {
                Width = 80,
                FlatStyle = FlatStyle.Flat,
                DataSource = Enum.GetValues(typeof(AggregateFunction)),
            };
            _cbAggregate.SelectedIndexChanged += _cbAggregate_SelectedIndexChanged;

            GroupBy = DateTimeGroupOption.ShowAll;
            Aggregate = AggregateFunction.Sum;

            this._pnlControlOptions.Controls.Add(_cbGroupBy);
            this._pnlControlOptions.Controls.Add(_cbAggregate);
        }
        #endregion

        #region Events
        public event EventHandler GroupByChanged;
        public event EventHandler AggregateChanged;
        #endregion

        #region Control EventHandlers
        private void _cbAggregate_SelectedIndexChanged(object sender, EventArgs e)
        {
            Aggregate = (AggregateFunction)_cbAggregate.SelectedValue;
            //AggregateChanged?.Invoke(sender, e);
        }
        private void _cbGroupBy_SelectedIndexChanged(object sender, EventArgs e)
        {
            GroupBy = (DateTimeGroupOption)_cbGroupBy.SelectedValue;
            //GroupByChanged?.Invoke(sender, e);
        }
        protected override void OnLoad(EventArgs e)
        {
            base.OnLoad(e);
            _cbGroupBy.SelectedIndex = (int)GroupBy;
            _cbAggregate.SelectedIndex = (int)Aggregate;
        }
        #endregion
    }
}

File: NetFramework/DataFilter/VB/ChartAsDataFilter/Controls/FlexChartDateTimeFilterView.vb
Imports C1.Chart

Namespace ChartAsDataFilter
    Partial Public Class FlexChartDateTimeFilterView
        Inherits FlexChartFilterView
#Region "fields"
        Private _cbGroupBy, _cbAggregate As ComboBox
        Private _groupBy As DateTimeGroupOption
        Private _aggregate As AggregateFunction
#End Region
#Region "Properties"
        Public Property GroupBy As DateTimeGroupOption
            Get
                Return _groupBy
            End Get
            Set(ByVal value As DateTimeGroupOption)

                If _groupBy <> value Then
                    _groupBy = value
                    _cbGroupBy.SelectedIndex = If(_cbGroupBy.SelectedIndex = -1, -1, CInt(value))
                    If GroupByChangedEvent IsNot Nothing Then RaiseEvent GroupByChanged(Me, Nothing)
                End If
            End Set
        End Property

        Public Property Aggregate As AggregateFunction
            Get
                Return _aggregate
            End Get
            Set(ByVal value As AggregateFunction)

                If _aggregate <> value Then
                    _aggregate = value
                    _cbAggregate.SelectedIndex = If(_cbAggregate.SelectedIndex = -1, -1, CInt(value))
                    If AggregateChangedEvent IsNot Nothing Then RaiseEvent AggregateChanged(Me, Nothing)
                End If
            End Set
        End Property

        Public Property ShowGroupBy As Boolean
            Get
                Return _cbGroupBy.Visible
            End Get
            Set(ByVal value As Boolean)
                _cbGroupBy.Visible = value
            End Set
        End Property

        Public Property ShowAggregate As Boolean
            Get
                Return _cbAggregate.Visible
            End Get
            Set(ByVal value As Boolean)
                _cbAggregate.Visible = value
            End Set
        End Property
#End Region
#Region "c'tor"
        Public Sub New(ByVal items As IEnumerable(Of Object), ByVal bindingX As String, ByVal binding As String, ByVal Optional chartType As ChartType = ChartType.Column)
            MyBase.New(items, bindingX, binding, chartType)
            InitializeComponent()
            _cbGroupBy = New ComboBox() With {
                .Width = 80,
                .FlatStyle = FlatStyle.Flat,
                .DataSource = [Enum].GetValues(GetType(DateTimeGroupOption))
            }
            AddHandler _cbGroupBy.SelectedIndexChanged, AddressOf _cbGroupBy_SelectedIndexChanged
            _cbAggregate = New ComboBox() With {
                .Width = 80,
                .FlatStyle = FlatStyle.Flat,
                .DataSource = [Enum].GetValues(GetType(AggregateFunction))
            }
            AddHandler _cbAggregate.SelectedIndexChanged, AddressOf _cbAggregate_SelectedIndexChanged
            GroupBy = DateTimeGroupOption.ShowAll
            Aggregate = AggregateFunction.Sum
            Me._pnlControlOptions.Controls.Add(_cbGroupBy)
            Me._pnlControlOptions.Controls.Add(_cbAggregate)
        End Sub
#End Region
#Region "Events"
        Public Event GroupByChanged As EventHandler
        Public Event AggregateChanged As EventHandler
#End Region
#Region "Control EventHandlers"
        Private Sub _cbAggregate_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs)
            Aggregate = CType(_cbAggregate.SelectedValue, AggregateFunction)
        End Sub

        Private Sub _cbGroupBy_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs)
            GroupBy = CType(_cbGroupBy.SelectedValue, DateTimeGroupOption)
        End Sub

        Protected Overrides Sub OnLoad(ByVal e As EventArgs)
            MyBase.OnLoad(e)
            _cbGroupBy.SelectedIndex = CInt(GroupBy)
            _cbAggregate.SelectedIndex = CInt(Aggregate)
        End Sub
#End Region
    End Class
End Namespace
